Can't build from main

Since today, I'm not able to build acquire-zarr from main. Details below. 

I am on Manjaro Linux, have installed pybind11 in my local development environment and cloned vcpkg following the README.

<details>

```
(acquire-dev) ➜  acquire-zarr git:(main) ✗ cmake --preset=default -B build -DBUILD_PYTHON=ON .
Preset CMake variables:

  CMAKE_TOOLCHAIN_FILE="vcpkg//scripts/buildsystems/vcpkg.cmake"

-- Running vcpkg install
Fetching registry information from https://github.com/microsoft/vcpkg (HEAD)...
Detecting compiler hash for triplet x64-linux...
Compiler found: /usr/bin/c++
The following packages will be built and installed:
    blosc:x64-linux@1.21.5 -- /home/melissa/.cache/vcpkg/registries/git-trees/7c0e6152db6058a53ff21fc8037e5c9e685a0fe8
    crc32c:x64-linux@1.1.2#2 -- /home/melissa/.cache/vcpkg/registries/git-trees/65cab652a6fffba3e3b25332d7935bf57aa3c4f0
  * curl[core,non-http,openssl,ssl]:x64-linux@8.8.0#1 -- /home/melissa/.cache/vcpkg/registries/git-trees/83d2bcc643fa6575f4cf536b69a2abb4cfa4f965
  * curlpp:x64-linux@2018-06-15#9 -- /home/melissa/.cache/vcpkg/registries/git-trees/5f1ce1e78bf1fca65c8f47bde61824589bd11e5c
  * inih[core,cpp]:x64-linux@58 -- /home/melissa/.cache/vcpkg/registries/git-trees/f86a866280aff2071eb313ff85cc1c5c7cabaeff
  * lz4:x64-linux@1.9.4#1 -- /home/melissa/.cache/vcpkg/registries/git-trees/d7704e3869f579ffdf927b5419579473e9d702d4
    minio-cpp:x64-linux@0.3.0 -- /home/melissa/.cache/vcpkg/registries/git-trees/5e77c0a5539cf25197ff13cb1eb5ad200c1c05ae
    nlohmann-json:x64-linux@3.11.3 -- /home/melissa/.cache/vcpkg/registries/git-trees/a728b86fa36d4885a4c57c79432660950644d6d4
  * openssl:x64-linux@3.3.1 -- /home/melissa/.cache/vcpkg/registries/git-trees/e117c9b91839ae0538b4e9a8d6809aecfa49948a
  * pugixml:x64-linux@1.14 -- /home/melissa/.cache/vcpkg/registries/git-trees/6e38344aea6e7529afde3895e0885ed5cb0c0542
  * snappy:x64-linux@1.1.10#1 -- /home/melissa/.cache/vcpkg/registries/git-trees/52d0b0f59c6f6e765c7f8df9e075ff7b90552c1e
  * vcpkg-cmake:x64-linux@2024-04-23 -- /home/melissa/.cache/vcpkg/registries/git-trees/e74aa1e8f93278a8e71372f1fa08c3df420eb840
  * vcpkg-cmake-config:x64-linux@2024-05-23 -- /home/melissa/.cache/vcpkg/registries/git-trees/97a63e4bc1a17422ffe4eff71da53b4b561a7841
  * vcpkg-cmake-get-vars:x64-linux@2023-12-31 -- /home/melissa/.cache/vcpkg/registries/git-trees/725c841810ceb8de33da68ea879eb8f0b4d6c94c
  * vcpkg-tool-meson:x64-linux@1.3.2#6 -- /home/melissa/.cache/vcpkg/registries/git-trees/13f79b5e4c8b4c07b38f9b5a010965433b753bb4
  * zlib:x64-linux@1.3.1 -- /home/melissa/.cache/vcpkg/registries/git-trees/3f05e04b9aededb96786a911a16193cdb711f0c9
  * zstd:x64-linux@1.5.6 -- /home/melissa/.cache/vcpkg/registries/git-trees/796171d4d359b6786b8e1380fb6da8e677a9087b
Additional packages (*) will be modified to complete this operation.
Restored 17 package(s) from /home/melissa/.cache/vcpkg/archives in 277 ms. Use --debug to see more details.
Installing 1/17 vcpkg-cmake-config:x64-linux@2024-05-23...
Elapsed time to handle vcpkg-cmake-config:x64-linux: 327 us
vcpkg-cmake-config:x64-linux package ABI: d45c0a9c01b8506fc726f1d9d7d932404c20bb4097f40543ee296f05686f76a0
Installing 2/17 vcpkg-cmake:x64-linux@2024-04-23...
Elapsed time to handle vcpkg-cmake:x64-linux: 277 us
vcpkg-cmake:x64-linux package ABI: ba506a1275e845216791af46ab19de8e59124680bf81ffd1851497b5746348b4
Installing 3/17 zstd:x64-linux@1.5.6...
Elapsed time to handle zstd:x64-linux: 553 us
zstd:x64-linux package ABI: 132f5ca4b3ee1a134cdc8a45517cf57780ce2570ea0fb8e252c1a3ffce23afae
Installing 4/17 zlib:x64-linux@1.3.1...
Elapsed time to handle zlib:x64-linux: 453 us
zlib:x64-linux package ABI: ce344fb05234ce1208db00c3d1ae56ab9a8d12480bf2ee2468f23e7162d29d28
Installing 5/17 snappy:x64-linux@1.1.10#1...
Elapsed time to handle snappy:x64-linux: 565 us
snappy:x64-linux package ABI: 6dea72b6262bffc681addc4c8ba6b0460b32d83ac61fd2e3308f4bd311953835
Installing 6/17 lz4:x64-linux@1.9.4#1...
Elapsed time to handle lz4:x64-linux: 684 us
lz4:x64-linux package ABI: a2a3d046c9a5c38790d77d038461036118324a88be0cfb27ffae3bd01ad40b20
Installing 7/17 blosc:x64-linux@1.21.5...
Elapsed time to handle blosc:x64-linux: 563 us
blosc:x64-linux package ABI: df23c18d65a1f167bb079fb7a11c080ec7629c9f2253540384838040eed20de5
Installing 8/17 crc32c:x64-linux@1.1.2#2...
Elapsed time to handle crc32c:x64-linux: 828 us
crc32c:x64-linux package ABI: eb6b8f5ea0c41c1b8346c4a516a151b2321a899f60eff401c773b3e660c36615
Installing 9/17 pugixml:x64-linux@1.14...
Elapsed time to handle pugixml:x64-linux: 1.03 ms
pugixml:x64-linux package ABI: fdde82340e2a29c190ff102614f5395626a56f3487a47e3b1d60be14e916fb2d
Installing 10/17 vcpkg-cmake-get-vars:x64-linux@2023-12-31...
Elapsed time to handle vcpkg-cmake-get-vars:x64-linux: 990 us
vcpkg-cmake-get-vars:x64-linux package ABI: 83060b3efb5ae694298577b3838285ff360bac0863bcdca1f33820286a7f77ee
Installing 11/17 openssl:x64-linux@3.3.1...
Elapsed time to handle openssl:x64-linux: 3.91 ms
openssl:x64-linux package ABI: 6276e427095b44bd7f35ef3b579e552aca2ffcdd436e8721ad6617f7951e8a0f
Installing 12/17 nlohmann-json:x64-linux@3.11.3...
Elapsed time to handle nlohmann-json:x64-linux: 1.91 ms
nlohmann-json:x64-linux package ABI: 6c66b84e41d34587873b03903a7e48d710bc3f8fc10c8fb5ba79e6ac610136c3
Installing 13/17 vcpkg-tool-meson:x64-linux@1.3.2#6...
Elapsed time to handle vcpkg-tool-meson:x64-linux: 854 us
vcpkg-tool-meson:x64-linux package ABI: 42cd0ce28536df10bba18d5fef69a2593fb21b0d0d28b6ca68d7a8cf4e6f5b01
Installing 14/17 inih[core,cpp]:x64-linux@58...
Elapsed time to handle inih:x64-linux: 835 us
inih:x64-linux package ABI: bdf496423def78ccbceea9150d9355b9fc8d65fb908dbf5d25cf98f6031e2f61
Installing 15/17 curl[core,non-http,openssl,ssl]:x64-linux@8.8.0#1...
Elapsed time to handle curl:x64-linux: 1.26 ms
curl:x64-linux package ABI: 953db4e66f11b6cd7d4279f4c17ea2c4d03ab4e0c4954de26e448670a18f96d4
Installing 16/17 curlpp:x64-linux@2018-06-15#9...
Elapsed time to handle curlpp:x64-linux: 1.32 ms
curlpp:x64-linux package ABI: 0f94b632c79057ef7dbbafa63389b163d2923ba4f1df7b174e8b90bdd4fcb35d
Installing 17/17 minio-cpp:x64-linux@0.3.0...
Elapsed time to handle minio-cpp:x64-linux: 952 us
minio-cpp:x64-linux package ABI: 2211d17489ae4b931ee79bb244898797b52d2dfc96cd74b143adbef54f388aa8
Total install time: 17.6 ms
blosc provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(blosc CONFIG REQUIRED)
  target_link_libraries(main PRIVATE blosc_static)

blosc provides pkg-config modules:

  # A blocking, shuffling and lossless compression library
  blosc

crc32c provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(Crc32c CONFIG REQUIRED)
  target_link_libraries(main PRIVATE Crc32c::crc32c)

The package nlohmann-json provides CMake targets:

    find_package(nlohmann_json CONFIG REQUIRED)
    target_link_libraries(main PRIVATE nlohmann_json::nlohmann_json)

The package nlohmann-json can be configured to not provide implicit conversions via a custom triplet file:

    set(nlohmann-json_IMPLICIT_CONVERSIONS OFF)

For more information, see the docs here:
    
    https://json.nlohmann.me/api/macros/json_use_implicit_conversions/

minio-cpp provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(miniocpp CONFIG REQUIRED)
  target_link_libraries(main PRIVATE miniocpp::miniocpp)

minio-cpp provides pkg-config modules:

  # MinIO C++ Client SDK provides simple APIs to access S3 compatible object storage
  miniocpp

-- Running vcpkg install - done
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found nlohmann_json: /home/melissa/projects/acquire-zarr/build/vcpkg_installed/x64-linux/share/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.11.3")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found ZLIB: /home/melissa/projects/acquire-zarr/build/vcpkg_installed/x64-linux/lib/libz.a (found version "1.3.1")
-- Found OpenSSL: /home/melissa/projects/acquire-zarr/build/vcpkg_installed/x64-linux/debug/lib/libcrypto.a (found version "3.3.1")
-- Found OpenSSL: /home/melissa/projects/acquire-zarr/build/vcpkg_installed/x64-linux/debug/lib/libcrypto.a (found suitable version "3.3.1", minimum required is "3")
-- Found ZLIB: /home/melissa/projects/acquire-zarr/build/vcpkg_installed/x64-linux/lib/libz.a (found suitable version "1.3.1", minimum required is "1")
-- Found ZLIB: /home/melissa/projects/acquire-zarr/build/vcpkg_installed/x64-linux/lib/libz.a (found version "1.3.1")
-- Version nightly dd44786
-- Skipping examples
-- Skipping benchmarks
-- Found Python3: /home/melissa/micromamba/envs/acquire-dev/bin/python3.11 (found version "3.11.10") found components: Interpreter Development Development.Module Development.Embed
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Found pybind11: /home/melissa/micromamba/envs/acquire-dev/lib/python3.11/site-packages/pybind11/include (found version "2.13.6")
-- Configuring done (5.0s)
-- Generating done (0.1s)
-- Build files have been written to: /home/melissa/projects/acquire-zarr/build
(acquire-dev) ➜  acquire-zarr git:(main) ✗ cmake --build build
[  1%] Building CXX object src/logger/CMakeFiles/acquire-logger-obj.dir/logger.cpp.o
[  1%] Built target acquire-logger-obj
[  2%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/acquire.zarr.cpp.o
In file included from /home/melissa/projects/acquire-zarr/src/streaming/acquire.zarr.cpp:3:
/home/melissa/projects/acquire-zarr/src/streaming/macros.hh:22:80: warning: barra invertida e nova linha no final do arquivo
   22 | #define EXPECT_VALID_INDEX(e, ...)                                             \
[  3%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/zarr.dimension.cpp.o
In file included from /home/melissa/projects/acquire-zarr/src/streaming/zarr.dimension.cpp:2:
/home/melissa/projects/acquire-zarr/src/streaming/macros.hh:22:80: warning: barra invertida e nova linha no final do arquivo
   22 | #define EXPECT_VALID_INDEX(e, ...)                                             \
[  4%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/zarr.stream.cpp.o
In file included from /home/melissa/projects/acquire-zarr/src/streaming/zarr.stream.cpp:1:
/home/melissa/projects/acquire-zarr/src/streaming/macros.hh:22:80: warning: barra invertida e nova linha no final do arquivo
   22 | #define EXPECT_VALID_INDEX(e, ...)                                             \
[  5%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/zarr.common.cpp.o
In file included from /home/melissa/projects/acquire-zarr/src/streaming/zarr.common.cpp:1:
/home/melissa/projects/acquire-zarr/src/streaming/macros.hh:22:80: warning: barra invertida e nova linha no final do arquivo
   22 | #define EXPECT_VALID_INDEX(e, ...)                                             \
[  7%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/blosc.compression.params.cpp.o
[  8%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/thread.pool.cpp.o
[  9%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/s3.connection.cpp.o
In file included from /home/melissa/projects/acquire-zarr/src/streaming/s3.connection.cpp:1:
/home/melissa/projects/acquire-zarr/src/streaming/macros.hh:22:80: warning: barra invertida e nova linha no final do arquivo
   22 | #define EXPECT_VALID_INDEX(e, ...)                                             \
[ 10%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/sink.cpp.o
In file included from /home/melissa/projects/acquire-zarr/src/streaming/sink.cpp:2:
/home/melissa/projects/acquire-zarr/src/streaming/macros.hh:22:80: warning: barra invertida e nova linha no final do arquivo
   22 | #define EXPECT_VALID_INDEX(e, ...)                                             \
[ 11%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/file.sink.cpp.o
[ 12%] Building CXX object src/streaming/CMakeFiles/acquire-zarr.dir/s3.sink.cpp.o
In file included from /home/melissa/projects/acquire-zarr/src/streaming/s3.sink.cpp:1:
/home/melissa/projects/acquire-zarr/src/streaming/macros.hh:22:80: warning: barra invertida e nova linha no final do arquivo
   22 | #define EXPECT_VALID_INDEX(e, ...)                                             \
/home/melissa/projects/acquire-zarr/src/streaming/s3.sink.cpp: In member function ‘virtual bool zarr::S3Sink::write(size_t, std::span<const std::byte>)’:
/home/melissa/projects/acquire-zarr/src/streaming/s3.sink.cpp:75:18: error: ‘copy_n’ is not a member of ‘std’; did you mean ‘copy’?
   75 |             std::copy_n(data_ptr,
      |                  ^~~~~~
      |                  copy
make[2]: *** [src/streaming/CMakeFiles/acquire-zarr.dir/build.make:205: src/streaming/CMakeFiles/acquire-zarr.dir/s3.sink.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1256: src/streaming/CMakeFiles/acquire-zarr.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
(acquire-dev) ➜  acquire-zarr git:(main) ✗ 
```

</details>

--- a/src/streaming/s3.sink.cpp
+++ b/src/streaming/s3.sink.cpp
@@ -1,5 +1,6 @@
 #include "macros.hh"
 #include "s3.sink.hh"
+#include <algorithm>
 
 #include <miniocpp/client.h>
 

